src/main/java/org/openrefine/extensions/commons/utils/CommonsImportingController.java

```diff
@@ -149,7 +150,7 @@ private void doParsePreview(
                 job.project,
                 job.metadata,
                 job,
-                DEFAULT_PREVIEW_LIMIT ,
+                DEFAULT_PROJECT_LIMIT,
```

Comment:
For a preview, I would expect to not fetch the whole category, so it would make sense to have a small limit. You should then make sure that the limit is disabled when the project is actually created.
If this code is run at project creation, then there is an architectural problem because this method is called `doParsePreview`.